Matrix.java

```diff
@@ -0,0 +1,210 @@
+class Matrix
```

Comment:
I think that both the class and the API that we have documentation for are meant to be public. Did you make them not public for a specific reason?